lib/mongoid/fields/field_types.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Fields
+
+    # Singleton module which contains a mapping of field types to field class.
+    # Custom field types can be configured.
+    module FieldTypes
+
+      # The default mapping of field type symbol to class.
+      #
+      # @api private
+      DEFAULT_MAPPING = {
+        array: Array,
+        bigdecimal: BigDecimal,
+        big_decimal: BigDecimal,
+        binary: BSON::Binary,
+        boolean: Mongoid::Boolean,
+        date: Date,
+        datetime: DateTime,
+        date_time: DateTime,
+        decimal128: BSON::Decimal128,
+        double: Float,
+        float: Float,
+        hash: Hash,
+        integer: Integer,
+        object: Object,
+        object_id: BSON::ObjectId,
+        range: Range,
+        regexp: Regexp,
+        set: Set,
+        string: String,
+        stringified_symbol: Mongoid::StringifiedSymbol,
+        symbol: Symbol,
+        time: Time,
+        time_with_zone: ActiveSupport::TimeWithZone
+      }.with_indifferent_access.freeze
+
+      class << self
+
+        # Gets a field class given its mapping identifier.
+        #
+        # @example
+        #   Mongoid::FieldTypes.get(:point)
+        #
+        # @param [ Symbol | String ] value the identifier of the defined type
+        def get(value)
+          mapping[value] || handle_unmapped_type(value)
+        end
+
+        # Defines a field type mapping, for later use in field :type option.
+        #
+        # @example
+        #   Mongoid::FieldTypes.define(:point, Point)
+        #
+        # @param [ Symbol ] symbol the symbol identifier of the defined type
+        # @param [ Class ] klass the class of the defined type, which must
+        #   include mongoize and related methods.
+        def define(symbol, klass)
+          mapping[symbol.to_sym] = klass
```

Comment:
Conversion to symbol is redundant because symbol is already a symbol.

Reply:
Done.

Reply:
Please mark resolved.